typescript-carousel/main.ts

```diff
@@ -0,0 +1,195 @@
+    $img2.className = 'hidden';
+    $img1.className = 'unhidden';
+    $dot1.className = 'filled';
+    $dot2.className = 'empty';
+  } else if ($img3.className === 'unhidden') {
+    $img3.className = 'hidden';
+    $img2.className = 'unhidden';
+    $dot2.className = 'filled';
+    $dot3.className = 'empty';
+  } else if ($img4.className === 'unhidden') {
+    $img4.className = 'hidden';
+    $img3.className = 'unhidden';
+    $dot3.className = 'filled';
+    $dot4.className = 'empty';
+  } else if ($img5.className === 'unhidden') {
+    $img5.className = 'hidden';
+    $img4.className = 'unhidden';
+    $dot4.className = 'filled';
+    $dot5.className = 'empty';
+  }
+}
+
+$right.addEventListener('click', function () {
+  imgSwitch();
+  clearInterval(intervalId);
+  startInterval();
+});
+
+$left.addEventListener('click', function () {
+  imgSwitchBack();
+  clearInterval(intervalId);
+  startInterval();
+});
+
+$dot1.addEventListener('click', function () {
+  $dot1.className = 'filled';
+  $img1.className = 'unhidden';
+  $dot2.className = 'empty';
+  $img2.className = 'hidden';
+  $dot3.className = 'empty';
+  $img3.className = 'hidden';
+  $dot4.className = 'empty';
+  $img4.className = 'hidden';
+  $dot5.className = 'empty';
+  $img5.className = 'hidden';
+  clearInterval(intervalId);
+  startInterval();
+});
+$dot2.addEventListener('click', function () {
+  $dot1.className = 'empty';
+  $img1.className = 'hidden';
+  $dot2.className = 'filled';
+  $img2.className = 'unhidden';
+  $dot3.className = 'empty';
+  $img3.className = 'hidden';
+  $dot4.className = 'empty';
+  $img4.className = 'hidden';
+  $dot5.className = 'empty';
+  $img5.className = 'hidden';
+  clearInterval(intervalId);
+  startInterval();
+});
+$dot3.addEventListener('click', function () {
+  $dot1.className = 'empty';
+  $img1.className = 'hidden';
+  $dot2.className = 'empty';
+  $img2.className = 'hidden';
+  $dot3.className = 'filled';
+  $img3.className = 'unhidden';
+  $dot4.className = 'empty';
+  $img4.className = 'hidden';
+  $dot5.className = 'empty';
+  $img5.className = 'hidden';
+  clearInterval(intervalId);
+  startInterval();
+});
+$dot4.addEventListener('click', function () {
+  $dot1.className = 'empty';
+  $img1.className = 'hidden';
+  $dot2.className = 'empty';
+  $img2.className = 'hidden';
+  $dot3.className = 'empty';
+  $img3.className = 'hidden';
+  $dot4.className = 'filled';
+  $img4.className = 'unhidden';
+  $dot5.className = 'empty';
+  $img5.className = 'hidden';
+  clearInterval(intervalId);
+  startInterval();
+});
+$dot5.addEventListener('click', function () {
+  $dot1.className = 'empty';
+  $img1.className = 'hidden';
+  $dot2.className = 'empty';
+  $img2.className = 'hidden';
+  $dot3.className = 'empty';
+  $img3.className = 'hidden';
+  $dot4.className = 'empty';
+  $img4.className = 'hidden';
+  $dot5.className = 'filled';
+  $img5.className = 'unhidden';
+  clearInterval(intervalId);
+  startInterval();
+});
```

Comment:
Same issue as an earlier comment with what happens if you add more or not.

Also don't add events to every single dot here! You should be using event delegation to have one event on the parent and delegation will allow the children to also be able to trigger the event!